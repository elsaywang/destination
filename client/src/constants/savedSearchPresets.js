export const topUnusedSignals = {
    advanced: false,
    descending: false,
    presetId: 'top-unused-signals',
    includeInDashboard: true,
    keyValuePairs: [{ id: 0, key: '', operator: '==', value: '' }],
    minEventFires: 1000,
    filterNewSignals: false,
    name: 'Top Unused Signals',
    signalStatus: 'UNUSED',
    sortBy: 'percentageChange',
    source: {
        dataSourceIds: [],
        name: '',
        reportSuiteIds: [],
        sourceType: 'ALL',
    },
    viewRecordsFor: '7D',
    includeSourceName: true,
};

export const newUnusedSignals = {
    ...topUnusedSignals,
    name: 'New Unused Signals',
    presetId: 'new-unused-signals',
    filterNewSignals: true,
};

export const savedSearchPresets = [topUnusedSignals, newUnusedSignals];
