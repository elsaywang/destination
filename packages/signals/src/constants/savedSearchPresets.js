export const topUnusedSignals = {
    advanced: false,
    presetId: 'top-unused-signals',
    includeInDashboard: true,
    keyValuePairs: [{ id: 0, key: '', operator: '==', value: '' }],
    minEventFires: 1000,
    filterNewSignals: false,
    name: 'Top Unused Signals',
    signalStatus: 'UNUSED',
    source: {
        dataSourceIds: [],
        name: '',
        reportSuiteIds: [],
        sourceType: 'ALL',
    },
    viewRecordsFor: '7D',
    sortBy: 'totalCount',
};

export const newUnusedSignals = {
    ...topUnusedSignals,
    name: 'New Unused Signals',
    presetId: 'new-unused-signals',
    filterNewSignals: true,
};

export const savedSearchPresets = [topUnusedSignals, newUnusedSignals];
