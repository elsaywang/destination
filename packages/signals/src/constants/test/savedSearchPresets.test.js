import { newUnusedSignals, savedSearchPresets } from '../savedSearchPresets';

describe('saved search presets constants', () => {
    it('matches snapshot', () => {
        expect(savedSearchPresets).toMatchSnapshot();
    });

    it('should be identified by a unique property, `presetId`', () => {
        const actual = savedSearchPresets.filter(
            savedSearch => typeof savedSearch.presetId !== 'undefined',
        );
        const expected = savedSearchPresets;

        expect(actual).toEqual(expected);
    });

    describe('New Unused Signals', () => {
        it('should contain `filterNewSignals: true`', () => {
            expect(newUnusedSignals.filterNewSignals).toBe(true);
        });
    });
});
