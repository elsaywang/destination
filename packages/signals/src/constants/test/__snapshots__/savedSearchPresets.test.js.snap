// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`saved search presets constants matches snapshot 1`] = `
Array [
  Object {
    "advanced": false,
    "filterNewSignals": false,
    "includeInDashboard": true,
    "keyValuePairs": Array [
      Object {
        "id": 0,
        "key": "",
        "operator": "==",
        "value": "",
      },
    ],
    "minEventFires": 1000,
    "name": "Top Unused Signals",
    "presetId": "top-unused-signals",
    "signalStatus": "UNUSED",
    "sortBy": "totalCount",
    "source": Object {
      "dataSourceIds": Array [],
      "name": "",
      "reportSuiteIds": Array [],
      "sourceType": "ALL",
    },
    "viewRecordsFor": "7D",
  },
  Object {
    "advanced": false,
    "filterNewSignals": true,
    "includeInDashboard": true,
    "keyValuePairs": Array [
      Object {
        "id": 0,
        "key": "",
        "operator": "==",
        "value": "",
      },
    ],
    "minEventFires": 1000,
    "name": "New Unused Signals",
    "presetId": "new-unused-signals",
    "signalStatus": "UNUSED",
    "sortBy": "totalCount",
    "source": Object {
      "dataSourceIds": Array [],
      "name": "",
      "reportSuiteIds": Array [],
      "sourceType": "ALL",
    },
    "viewRecordsFor": "7D",
  },
]
`;
